addons/docs/src/frameworks/web-components/sourceDecorator.ts

```diff
@@ -0,0 +1,37 @@
+/* global window */
+import { render } from 'lit-html';
+import { addons, StoryContext, StoryFn } from '@storybook/addons';
+import { SNIPPET_RENDERED, SourceType } from '../../shared';
+
+function skipSourceRender(context: StoryContext) {
+  const sourceParams = context?.parameters.docs?.source;
+  const isArgsStory = context?.parameters.__isArgsStory;
+
+  // always render if the user forces it
+  if (sourceParams?.type === SourceType.DYNAMIC) {
+    return false;
+  }
+
+  // never render if the user is forcing the block to render code, or
+  // if the user provides code, or if it's not an args story.
+  return !isArgsStory || sourceParams?.code || sourceParams?.type === SourceType.CODE;
+}
+
+function applyTransformSource(source: string, context: StoryContext): string {
+  const { transformSource } = context.parameters.docs ?? {};
+  if (typeof transformSource !== 'function') return source;
+  return transformSource(source, context);
+}
+
+export function sourceDecorator(storyFn: StoryFn, context: StoryContext) {
+  const story = storyFn();
```

Comment:
How do others feel about `context.originalStoryFn` here?

Reply:
Here are my thoughts on the matter:
https://github.com/storybookjs/storybook/pull/14652#issuecomment-846549496

Since this is all brand new in web-components, we can make it opt-out OR opt-in, depending on how you guys feel. I believe it's safer to include the decorators by default and allow users to opt-out of the decorators. In the React world, it's quite common for stories to rely on the decorators to provide context to render properly. But I'll defer to your collective judgment for what's best in the web-components world.

Reply:
@Westbrook can you elaborate on the case, both ways?

Reply:
Yeah, I’m less worried as long as it’s configurable. However, for my library I use a very Storybook custom decorator to control theme color/scale/direction in a way that you wouldn’t want to require a copy/paster to clean up after the fact. `storyFn()` leaves all of the decorator code whereas without it you copy just the component under demo.

Reply:
@bennypowers @Westbrook WDYT about adding configuration in this PR. i'm afraid if it doesn't get done now it will slip through the cracks.

Reply:
I'm happy to put this in i'm just unfamiliar with the relevant apis.

is it just
```ts
const story = context.stripDecorators ? context.originalStoryFn() : storyFn(): 
```

??

Reply:
Let's make it consistent with React:

```js
const story = context?.parameters.docs?.source?.excludeDecorators
    ? context.originalStoryFn(context.args) 
    : storyFn();
```

Reply:
Added with unit tests, thanks